django_cloud_deploy/cli/prompt_manager.py

```diff
@@ -0,0 +1,689 @@
+import os.path
+import random
+import re
+import string
+import time
+from typing import Any, Dict, List, Optional
+import webbrowser
+
+from django_cloud_deploy import workflow
+from django_cloud_deploy.cli import io
+from django_cloud_deploy.cli import prompt_new
+from django_cloud_deploy.cloudlib import auth
+from django_cloud_deploy.cloudlib import billing
+from django_cloud_deploy.cloudlib import project
+
+
+class TemplatePromptManager(object):
+    """Used as a base template for all Prompt Managers interacting with user.
+
+    They should all have prompt method that should make subsequent prompt calls.
+    Manager should own only one parameter, for which they should have a
+    validate function for which they will call if the argument is already
+    supplied. They also handle any busines logic related to a prompt, as well
+    as become instantiated with any API related classes.
+
+    Args:
+        console: Object to use for user I/O.
+        step: Message to present to user regarding what step they are on.
+        args: Dictionary holding prompts answered by user and set up arguments.
+    """
+
+    # Parameter must be set for dictionary key
+    PARAMETER = None
+
+    def __init__(self, console: io.IO, step: str, args):
+        self.console = console
+        self.step = step
+        self.args = args
+
+    def prompt(self) -> None:
+        """Handles the business logic and calls the prompts."""
+        pass
+
+    def validate(self, val: str):
+        """Checks if the string is valid. Throws a ValueError if not valid."""
+        return
+
+    def get_credentials(self):
```

Comment:
Consider an `@property` in Python rather than a getter method.  if there is a need for this to be overridden, `@property` for `.credentials` that just calls a `_get_credentials` method that intended to be overidden in subclasses.